pkg/bgp/speaker/speaker.go

```diff
@@ -423,36 +423,54 @@ func convertEndpointSliceV1(in *slim_discover_v1.EndpointSlice) *metallbspr.Endp
 	}
 	out := new(metallbspr.Endpoints)
 	for _, ep := range in.Endpoints {
-		for _, addr := range ep.Addresses {
-			out.Ready = append(out.Ready, metallbspr.Endpoint{
-				IP:       addr,
-				NodeName: ep.NodeName,
-			})
+		if isConditionReadyForSliceV1(ep.Conditions) {
+			for _, addr := range ep.Addresses {
+				out.Ready = append(out.Ready, metallbspr.Endpoint{
+					IP:       addr,
+					NodeName: ep.NodeName,
+				})
+			}
 		}
 		// See above comment in convertEndpoints() for why we only append
 		// "ready" endpoints.
 	}
 	return out
 }
 
+func isConditionReadyForSliceV1(conditions slim_discover_v1.EndpointConditions) bool {
```

Comment:
I think naming could be made a bit more explicit here, for example what about `endpointIsReady()` (which would simply takes a `*slim_discover_v1.Endpoint` object)?